interface/ceed-qfunction.c

```diff
@@ -602,7 +602,11 @@ int CeedQFunctionCreateInterior(Ceed ceed, CeedInt vec_length,
   (*qf)->is_context_writable = true;
   (*qf)->function = f;
   (*qf)->user_flop_estimate = -1;
-  if (strlen(source)) {
+  // TODO: This clumsily assumes no JIT on host to avoid searching the file
+  // system. Searching the filesystem fails with WASM (default wasmer run).
+  CeedMemType mem_type;
+  ierr = CeedGetPreferredMemType(ceed, &mem_type); CeedChk(ierr);
+  if (strlen(source) && mem_type != CEED_MEM_HOST) {
     bool is_absolute_path;
     char *absolute_path;
```

Comment:
@jeremylt How concerned are you about this hack? It needs a tweak to fix the tests (example below), but I'm not sure whether to add an explicit field to specify *not* to check the filesystem or to infer from the environment/gracefully fall back when it can't.
```diff
--- tests/output/t402-qfunction.out
+++ New
@@ -1,4 +1,4 @@
-User CeedQFunction - setup
+User CeedQFunction - (null)
   1 input field:
     Input field 0:
       Name: "w"
@@ -9,7 +9,7 @@
       Name: "qdata"
       Size: 1
       EvalMode: "none"
-User CeedQFunction - mass
+User CeedQFunction - (null)
   2 input fields:
     Input field 0:
       Name: "qdata"
```

Reply:
It would be nice to preserve this piece of debugging info if it is reasonably possible.

Reply:
We have the name (I just need a more careful conditional), but `CeedGetJitAbsolutePath` fails under WASM so we need to skip that. But what should we use to decide whether to try? `mem_type`, some other environment info, or always try but with graceful failure (which might hide a different issue when run in an environment that should be able to find it)?

Reply:
We could go for a more lazy approach. Always store the full string and the name but only form and store the abspath if the backend requests it?

Reply:
That sounds good.